### PR TITLE
Split global state

### PR DESCRIPTION
**Pull request by RunDevelopment**
This PR does a bunch of things around `GlobalContext`:

- Split `GlobalContext` into `GlobalContext` (mostly never-changing stuff) and `GlobalVolitileContext` (stuff that changes all the time (nodes, edges))
- Changed some APIs and returned context values to make them more memoizable.
- Use [`use-context-selector`](https://github.com/dai-shi/use-context-selector) for contexts to avoid re-rendering.
- Changed how input elements get their data to avoid re-rendering.
- More efficient way to store nodes and edges in session storage to avoid re-rendering.

### CODE REVIEW

File: src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx [hunk 1]
```diff
@@ -18,10 +19,9 @@ interface RepresentativeNodeWrapperProps {
 }
 
 const RepresentativeNodeWrapper = ({ node }: RepresentativeNodeWrapperProps) => {
-    const { createNode, reactFlowInstance, reactFlowWrapper, useHoveredNode } =
-        useContext(GlobalContext);
-
-    const [, setHoveredNode] = useHoveredNode;
+    const createNode = useContextSelector(GlobalChainContext, (c) => c.createNode);
+    const reactFlowInstance = useContextSelector(GlobalChainContext, (c) => c.reactFlowInstance);
```
Comment (joeyballentine) [inline src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx:23]: Didn't we get rid of reactFlowInstance? Is this still needed?

File: src/helpers/contexts/GlobalNodeState.tsx [hunk 2]
```diff
@@ -178,28 +173,23 @@ export const GlobalProvider = ({
     schemata,
     reactFlowWrapper,
 }: React.PropsWithChildren<GlobalProviderProps>) => {
-    const { useSnapToGrid } = useContext(SettingsContext);
+    const useSnapToGrid = useContextSelector(SettingsContext, (c) => c.useSnapToGrid);
 
-    const [nodes, setNodes, onNodesChange] = useNodesState<NodeData>([]);
-    const [edges, setEdges, onEdgesChange] = useEdgesState<EdgeData>([]);
```
Comment (joeyballentine) [inline src/helpers/contexts/GlobalNodeState.tsx:184]: Does everything still work without the onNodes/EdgesChange functions?

File: src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx (hunk 1 above)
Comment (joeyballentine) [reply src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx:23]: After doing a code search I've determined that reactFlowInstance is not needed and can be removed.

File: src/helpers/contexts/GlobalNodeState.tsx (hunk 2 above)
Comment (RunDevelopment) [reply src/helpers/contexts/GlobalNodeState.tsx:184]: They weren't used at all. That's why I removed them.

File: src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx (hunk 1 above)
Comment (RunDevelopment) [reply src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx]: There are two instances where we use `reactFlowInstance.project()`. How would those be replaced?

Comment (joeyballentine) [reply src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx]: That can be replaced with the new (in v10) useReactFlow hook which contains the project method https://reactflow.dev/docs/api/hooks/use-react-flow/ https://reactflow.dev/docs/api/react-flow-instance/

File: src/helpers/contexts/GlobalNodeState.tsx (hunk 2 above)
Comment (joeyballentine) [reply src/helpers/contexts/GlobalNodeState.tsx:184]: Oh, I guess the ones in ReactFlowBox do all the work. Makes sense.

File: src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx (hunk 1 above)
Comment (RunDevelopment) [reply src/components/NodeSelectorPanel/RepresentativeNodeWrapper.tsx]: Good to know.

**Pull request merged**